miku-server/src/main/scala/org/http4s/miku/MikuBuilder.scala

```diff
@@ -0,0 +1,492 @@
+    copy(banner = banner)
+
+  def withIdleTimeout(idleTimeout: Duration): MikuBuilder[F] =
+    copy(idleTimeout = idleTimeout)
+
+  def withSSL(
+      keyStore: SSLKeyStoreSupport.StoreInfo,
+      keyManagerPassword: String,
+      protocol: String,
+      trustStore: Option[SSLKeyStoreSupport.StoreInfo],
+      clientAuth: Boolean
+  ): MikuBuilder[F] = {
+    val auth = if (clientAuth) ClientAuthRequired else NoClientAuth
+    val bits = MikuKeyStoreBits(keyStore, keyManagerPassword, protocol, trustStore, auth)
+    copy(sslBits = Some(bits))
+  }
+
+  def withSSLContext(sslContext: SSLContext, clientAuth: Boolean): MikuBuilder[F] = {
+    val auth = if (clientAuth) ClientAuthRequired else NoClientAuth
+    copy(sslBits = Some(MikuSSLContextBits(sslContext, auth)))
+  }
+
+  def withWebSockets(enableWebsockets: Boolean): MikuBuilder[F] =
+    copy(enableWebsockets = enableWebsockets)
+
+  /** Netty-only option **/
+  def withTransport(transport: NettyTransport): MikuBuilder[F] =
+    copy(transport = transport)
+
+  /** Netty-only option **/
+  def withChannelOptions(channelOptions: MikuBuilder.NettyChannelOptions) =
+    copy(nettyChannelOptions = channelOptions)
+
+  protected[this] def newRequestHandler(): ChannelInboundHandler =
+    if (enableWebsockets)
+      MikuHandler.websocket[F](httpService, serviceErrorHandler)
+    else
+      MikuHandler.default[F](httpService, serviceErrorHandler)
+
+  /** Return our SSL context.
+    *
+    */
+  private[this] def getContext(): Option[(SSLContext, ClientAuth)] = sslBits.map {
+    case MikuKeyStoreBits(keyStore, keyManagerPassword, protocol, trustStore, clientAuth) =>
+      val ksStream = new FileInputStream(keyStore.path)
+      val ks = KeyStore.getInstance("JKS")
+      ks.load(ksStream, keyStore.password.toCharArray)
+      ksStream.close()
+
+      val tmf = trustStore.map { auth =>
+        val ksStream = new FileInputStream(auth.path)
+
+        val ks = KeyStore.getInstance("JKS")
+        ks.load(ksStream, auth.password.toCharArray)
+        ksStream.close()
+
+        val tmf = TrustManagerFactory.getInstance(TrustManagerFactory.getDefaultAlgorithm)
+
+        tmf.init(ks)
+        tmf.getTrustManagers
+      }
+
+      val kmf = KeyManagerFactory.getInstance(
+        Option(Security.getProperty("ssl.KeyManagerFactory.algorithm"))
+          .getOrElse(KeyManagerFactory.getDefaultAlgorithm)
+      )
+
+      kmf.init(ks, keyManagerPassword.toCharArray)
+
+      val context = SSLContext.getInstance(protocol)
+      context.init(kmf.getKeyManagers, tmf.orNull, null)
+
+      (context, clientAuth)
+
+    case MikuSSLContextBits(context, clientAuth) =>
+      (context, clientAuth)
+  }
+
+  /** A stream transformation that registers our channels and
+    * adds the necessary codecs and transport handlers
+    */
+  def channelPipe(
+      eventLoop: MultithreadEventLoopGroup,
+      allChannels: DefaultChannelGroup): Pipe[F, Channel, Unit] = { s =>
```

Comment:
Does this get anything out of being a `Pipe` rather than a `Channel => F[Unit]`?

Reply:
Absolutely nothing I'm sure. This is a Pipe since it was a translation of play's equivalent of this in akka streams. I didn't give it much thought after initial translation

Reply:
I don't think it hurts anything.  It just looks like more power than you need.